src/mz-debug/src/kubectl_port_forwarder.rs

```diff
@@ -131,41 +131,47 @@ pub struct ServiceInfo {
 /// Returns ServiceInfo for balancerd
 pub async fn find_environmentd_service(
     client: &Client,
-    k8s_namespaces: &Vec<String>,
+    k8s_namespace: &String,
+    mz_instance_name: &String,
 ) -> Result<ServiceInfo> {
-    for namespace in k8s_namespaces {
-        let services: Api<Service> = Api::namespaced(client.clone(), namespace);
-        let services = services
-            .list(&ListParams::default().labels("materialize.cloud/mz-resource-id"))
-            .await
-            .with_context(|| format!("Failed to list services in namespace {}", namespace))?;
-
-        // Find the first sql service that contains balancerd
-        let maybe_service =
-            services
-                .iter()
-                .find_map(|service| match (&service.metadata.name, &service.spec) {
-                    (Some(service_name), Some(spec)) => {
-                        if !service_name.to_lowercase().contains("environmentd") {
-                            return None;
-                        }
+    let services_api: Api<Service> = Api::namespaced(client.clone(), k8s_namespace);
 
-                        if let Some(ports) = &spec.ports {
-                            Some(ServiceInfo {
-                                service_name: service_name.clone(),
-                                service_ports: ports.clone(),
-                                namespace: namespace.clone(),
-                            })
-                        } else {
-                            None
-                        }
+    let label_filter = format!(
+        // mz-resource-id is used to identify environmentd services
+        "materialize.cloud/mz-resource-id,materialize.cloud/organization-name={}",
+        mz_instance_name
+    );
+
+    let services = services_api
+        .list(&ListParams::default().labels(&label_filter))
+        .await
+        .with_context(|| format!("Failed to list services in namespace {}", k8s_namespace))?;
+
+    // Find the first sql service that contains balancerd
+    let maybe_service =
+        services
+            .iter()
+            .find_map(|service| match (&service.metadata.name, &service.spec) {
+                (Some(service_name), Some(spec)) => {
+                    if !service_name.to_lowercase().contains("environmentd") {
```

Comment:
Not something to fix in this PR, but I am realizing now that this could match both the generation service and the globally active one. It is unclear which we are looking for in this function, but I would guess we want the active one.